aiohttp/web_routedef.py

```diff
@@ -191,3 +192,28 @@ def view(self, path: str, **kwargs: Any) -> _Deco:
     def static(self, prefix: str, path: PathLike,
                **kwargs: Any) -> None:
         self._items.append(StaticDef(prefix, path, kwargs))
+
+
+class ClassRouteTableDef(RouteTableDef):
+    def __repr__(self) -> str:
+        return "<ClassRouteTableDef count={}>".format(len(self._items))
+
+    def route(self,
+              method: str,
+              path: str,
+              **kwargs: Any) -> _Deco:
+        def inner(handler: Any) -> Any:
+            # Is Any rather than _HandlerType because of mypy limitations, see
+            # https://github.com/python/mypy/issues/3882 and
+            # https://github.com/python/mypy/issues/2087
+            handler.route_info = (method, path, kwargs)
+            return handler
+        return inner
+
+    def add_class_routes(self, instance: Any) -> None:
+        def predicate(member: Any) -> bool:
+            return all((inspect.iscoroutinefunction(member),
+                        hasattr(member, "route_info")))
+        for _, handler in inspect.getmembers(instance, predicate):
+            method, path, kwargs = handler.route_info
+            super().route(method, path, **kwargs)(handler)
```

Comment:
The code is cumbersome.
You override `route()` method a few lines above but calls a parent version here.
It confused me on code review, I believe any next code reader will stumble on this line as well.
Please find a way to avoid such magic.

Reply:
I really don't know how I could avoid it without having to duplicate code (creating problems) or introducing a new name for the subclass's @route decorator, which would be really confusing for the user.